Composer/packages/client/src/components/CreateSkillModal.tsx

```diff
@@ -1,265 +1,279 @@
+  } else if (!urlRegex.test(endpointUrl) || !msAppIdRegex.test(msAppId)) {
+    setFormDataErrors({ ...errors, endpoint: formatMessage('Skill manifest endpoint is configured improperly') });
+  } else {
+    setFormDataErrors(errors);
+    setValidationState({ ...validationState, endpoint: ValidationState.Validated });
+  }
+};
 
-      if (currentError) {
-        errors.manifestUrl = currentError;
-      } else {
-        delete errors.manifestUrl;
-      }
+export const validateManifestUrl = async ({
+  formData,
+  formDataErrors,
+  projectId,
+  skills,
+  setFormDataErrors,
+  setValidationState,
+  setSkillManifest,
+  validationState,
+}) => {
+  const { manifestUrl } = formData;
+  const { manifestUrl: _, ...errors } = formDataErrors;
+
+  if (!manifestUrl) {
+    setFormDataErrors({ ...errors, manifestUrl: formatMessage('Please input a manifest Url') });
+  } else if (!urlRegex.test(manifestUrl)) {
+    setFormDataErrors({ ...errors, manifestUrl: formatMessage('Url should start with http[s]://') });
+  } else if (skills.some((skill) => skill.manifestUrl === manifestUrl)) {
+    setFormDataErrors({ ...errors, manifestUrl: formatMessage('Duplicate skill manifest Url') });
+  } else {
+    try {
+      setValidationState({ ...validationState, manifestUrl: ValidationState.Validating });
+      const { data } = await httpClient.post(`/projects/${projectId}/skill/check`, { url: manifestUrl });
+      setFormDataErrors(errors);
+      setSkillManifest(data);
+      setValidationState({ ...validationState, manifestUrl: ValidationState.Validated });
+    } catch (error) {
+      setFormDataErrors({ ...errors, manifestUrl: formatMessage('Manifest url can not be accessed') });
+      setValidationState({ ...validationState, manifestUrl: ValidationState.NotValidated });
     }
+  }
+};
 
-    if (newData.name != null) {
-      const { name } = newData;
-      let currentError = '';
-      if (name) {
-        if (!skillNameRegex.test(name)) {
-          currentError = formatMessage('Name contains invalid charactors');
-        }
-
-        const duplicatedItemIndex = skills.findIndex((item) => item.name === name);
-        if (duplicatedItemIndex !== -1 && (!isModify || (isModify && duplicatedItemIndex !== editIndex))) {
-          currentError = formatMessage('Duplicate skill name');
-        }
-      }
-
-      if (currentError) {
-        errors.name = currentError;
-      } else {
-        delete errors.name;
-      }
-    }
+export const validateName = ({
+  formData,
+  formDataErrors,
+  skills,
+  setFormDataErrors,
+  setValidationState,
+  validationState,
+}) => {
+  const { name } = formData;
+  const { name: _, ...errors } = formDataErrors;
+
+  if (name && !skillNameRegex.test(name)) {
+    setFormDataErrors({ ...errors, name: formatMessage('Name cannot include special characters or spaces') });
+  } else if (name && skills.some((skill) => skill.name === name)) {
```

Comment:
Do you think we should allow same skill name with different case? Hello-skill and hello-skill. I feel like we should'nt

Reply:
I agree. We should probably do the same thing for the manifest url.